ipq40xx: fix and refactor ZTE MF287 series

andyboeh opened the pull request:
The ZTE MF287 requires a different board calibration file for ath10k than the ZTE MF287+. The two devices receive their own DTS, thus the device tree is slightly refactored.

While at it, turn the "gpio-restart" node into a "gpio-export" similar to the MF287 Pro. Unfortunately, there doesn't seem to be a "power button blocker" GPIO, so a modem reset always triggers a system reset.

This needs to be backported to 23.05 as the MF287 builds have broken WiFi.

andyboeh committed:
ipq40xx: refactor ZTE MF287 series

The ZTE MF287 requires a different board calibration file for ath10k than
the ZTE MF287+. The two devices receive their own DTS, thus the device tree
is slightly refactored.

Signed-off-by: Andreas Böhler <dev@aboehler.at>

robimarko commented:
Is it possible to split these changes into multiple commits?
Ideally, fixes and improvements should be separate, as this is quite hard to review currently.

andyboeh committed:
ipq40xx: ZTE MF287 series: move to gpio-export for modem-reset GPIO

Turn the "gpio-restart" node into a "gpio-export" node for all MF287
variants, similar to the MF287 Pro. Unfortunately, there doesn't seem to be
a "power button blocker" GPIO for the MF287 and MF287 Plus, so a modem
reset always triggers a system reset.

Signed-off-by: Andreas Böhler <dev@aboehler.at>

andyboeh commented:
Sure, I turned it into a refactor-commit and an improvement commit

andyboeh commented:
@Ansuel thanks, shall I cherry-pick the commits and create a backport-PR for 23.05?
